app/models/user.rb

```diff
@@ -0,0 +1,23 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+
+  validates :email, presence: true,
+    length: {maximum: Settings.maximum_email_length},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+
+  validates :name,  presence: true,
+    length: {maximum: Settings.maximum_name_length}
+
+  validates :password, presence: true,
+    length: {minimum: Settings.minimum_password_length}
+
+  before_save{email_downcase}
+
+  has_secure_password
+
+  private
+  def email_downcase
```

Comment:
có cách nào để anh gọi đc method này ở ngoài không

Reply:
Một số cách theo em nghĩ ạ:
+) dùng toán tử gán trực tiếp  seft.email=email.downcase
+) Dùng "bang" method (cái dấu !): email.downcase!
+) method này có thể được khai báo class khác và gọi được nó thì phương thức phải được khai báo là public